rosbag2_storage_default_plugins/src/rosbag2_storage_default_plugins/sqlite/sqlite_storage.cpp

```diff
@@ -73,15 +67,15 @@ void SqliteStorage::open(
     relative_path_ = uri + FILE_EXTENSION;
 
     // READ_WRITE requires the DB to not exist.
-    if (database_exists(relative_path_)) {
+    if (rosbag2_storage::FilesystemHelper::file_exists(relative_path_)) {
       throw std::runtime_error(
               "Failed to create bag: File '" + relative_path_ + "' already exists!");
     }
   } else {  // APPEND and READ_ONLY
     relative_path_ = uri;
 
     // APPEND and READ_ONLY require the DB to exist
-    if (!database_exists(relative_path_)) {
+    if (!rosbag2_storage::FilesystemHelper::file_exists(relative_path_)) {
```

Comment:
If I put in a valid file, which doesn't point to a `.db3` file, this check will still return true. Do you think it makes sense to check for a `.db3` suffix?

Reply:
I'm unsure on this one. Sqlite3 doesn't actually care what the extension is and Line 78 is enforcing that the database exists. The line I added to `SqliteWrapper` constructor throws when the database cannot be opened or if its invalid, so validity is being checked.

Do you think, from a design point of view, that `SqliteStorage` should only be allowed to open databases whose `uri` ends with `db3`?

Reply:
i guess that's okay then.